src/hooks/Tokens.ts

```diff
@@ -5,42 +5,49 @@ import { SupportedChainId } from 'constants/chains'
 import { DEFAULT_INACTIVE_LIST_URLS, DEFAULT_LIST_OF_LISTS } from 'constants/lists'
 import { useCurrencyFromMap, useTokenFromMapOrNetwork } from 'lib/hooks/useCurrency'
 import { getTokenFilter } from 'lib/hooks/useTokenList/filtering'
+import { ChainTokenInfoMap } from 'lib/hooks/useTokenList/utils'
 import { useMemo } from 'react'
 import { isL2ChainId } from 'utils/chains'
 
 import { useAllLists, useCombinedActiveList, useCombinedTokenMapFromUrls } from '../state/lists/hooks'
 import { WrappedTokenInfo } from '../state/lists/wrappedTokenInfo'
 import { useUserAddedTokens, useUserAddedTokensOnChain } from '../state/user/hooks'
-import { TokenAddressMap, useUnsupportedTokenList } from './../state/lists/hooks'
+import { useUnsupportedTokenList } from './../state/lists/hooks'
 
-// reduce token map into standard address <-> Token mapping, optionally include user added tokens
-function useTokensFromMap(tokenMap: TokenAddressMap): { [address: string]: Token } {
+export type TokenMap = { [address: string]: Token | undefined }
+export function tokenMapToArray(tokens: TokenMap): Array<Token> {
+  return Object.values(tokens).filter((token): token is Token => token !== undefined)
```

Comment:
Can you just use `Boolean` here?

```suggestion
  return Object.values(tokens).filter<Token>(Boolean)
```

Reply:
Boolean filtering doesn't appease the type-checker; some form of cast/guard is needed.  do you think
```suggestion
  return Object.values(tokens).filter(Boolean) as Array<Token>
```
is better?